bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -63,12 +69,13 @@
 import org.apache.bookkeeper.util.SafeRunnable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.apache.bookkeeper.client.api.WriteHandle;
 
 /**
  * Ledger handle contains ledger metadata and is used to access the read and
  * write operations to a ledger.
  */
-public class LedgerHandle implements AutoCloseable {
+public class LedgerHandle implements AutoCloseable, WriteHandle {
```

Comment:
hmm..not all LedgerHandles can be WriteHandle , right?

Reply:
LedgerHandle, LedgerHandleAdv and ReadonlyLedgerHandle is a really messy class hierarchy.
For new clients which will use the new org.apache.bookkeeper.client.api package this mess will be hidden. Actually ReadonlyLedgerHandle extends LedgerHandle